pkg/networks/usernet/gvproxy.go

```diff
@@ -56,21 +63,21 @@ func StartGVisorNetstack(ctx context.Context, gVisorOpts *GVisorNetstackOpts) er
 	config := types.Configuration{
 		Debug:             false,
 		MTU:               opts.MTU,
-		Subnet:            networks.SlirpNetwork,
-		GatewayIP:         networks.SlirpDNS,
+		Subnet:            opts.Subnet,
+		GatewayIP:         gatewayIP,
```

Comment:
This fixes the in-consistency mentioned here.
https://github.com/lima-vm/lima/discussions/1600

DNS requests are handled by gateway itself/forwarded to host automatically. So there is no need for a explicit use of DNS IP.